src/PythonMigrationViewExtension/MigrationAssistant/ScriptMigrator.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Python.Runtime;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Dynamo.PythonMigration.MigrationAssistant
+{
+    internal static class ScriptMigrator
+    {
+        internal static string MigrateCode(List<string> inputNames, List<object> inputValues, string returnName)
+        {
+            Python.Included.Installer.SetupPython().Wait();
+
+            if (!PythonEngine.IsInitialized)
+            {
+                PythonEngine.Initialize();
+                PythonEngine.BeginAllowThreads();
+            }
+
+            IntPtr gs = PythonEngine.AcquireLock();
+
+            try
+            {
+                using (Py.GIL())
+                {
+
+                    using (PyScope scope = Py.CreateScope())
+                    {
+                        int amt = Math.Min(inputNames.Count, inputValues.Count);
+
+                        for (int i = 0; i < amt; i++)
+                        {
+                            scope.Set(inputNames[i], inputValues[i].ToPython());
+                        }
+
+                        try
+                        {
+                            scope.Exec(GetPythonMigrationScript());
```

Comment:
if this python script encounters an error
- does it throw here
- and do we have access to the error message to bubble it further up to the UI for user to see ?